### DISCUSSION TITLE
How to use this library on Kotlin?

### DISCUSSION THREAD

**Discussion by mkutr** [CoderLine/alphaTab#1134] [Q&A] [answered]
As in the title.
There are guides for Net and JS, there are no guides for kotlin. Couldn't find the library on maven central. Should I compile a jar locally? How to do it? What java version to use? How to set up the compilation? 

I know I can run `npm run build-kotlin` but it leads to wierd errors (for example the generated code reassigns values to parameters: `e: dist/lib.kotlin/commonMain/generated/synth/ds/CircularSampleBuffer.kt:107:13 Val cannot be reassigned`
) and this doesn't compile properly.

**Comment by Danielku15** [comment, marked as answer]
Kotlin is an upcoming feature in the **future** 1.3 release. That's why official & stable libraries are not available *yet* and the related guides on the website are also an outstanding task. You can get an overview on that here: https://github.com/orgs/CoderLine/projects/13

But(!): We are currently publishing already 1.3 prerelease versions to the Sonatype OSS Maven repositories: https://s01.oss.sonatype.org/content/repositories/snapshots/net/alphatab/ hence you should be able to consume alphaTab (core) or alphaTab (android) already via Maven and Gradle: https://central.sonatype.org/consume/ 
Since a few days the publish seems broken and I'll have to fix that but there should be fairly up-to-date builds available. 
The available APIs and codebase is similar to the other platforms, so you depending on your experience you might find your way as only the syntax is different. You might also check out https://github.com/CoderLine/alphaTabSamplesAndroid/tree/main/src where already some early example is available but at this point I cannot guarantee full stability. 

Regarding the compilation problem: Can you share some insights on your development environment? Generally things compile fine on my environment and GitHub. I would rather expect that compilation works everywhere.
